src/EnergyPlus/MixedAir.cc

```diff
@@ -4655,7 +4655,9 @@ namespace MixedAir {
                     AirLoopControlInfo(AirLoopNum).HeatRecoveryBypass = true;
                     this->HeatRecoveryBypassStatus = 1;
                 } else if (this->HeatRecoveryBypassControlType == BypassWhenOAFlowGreaterThanMinimum) {
-                    if (OASignal > OutAirMinFrac) {
+                    Real64 OAMassFlowMin = OutAirMinFrac * AirLoopFlow(AirLoopNum).DesSupply;
+                    Real64 OAMassFlowActual = OASignal * this->MixMassFlow;
+                    if (OAMassFlowActual > OAMassFlowMin) {
                         AirLoopControlInfo(AirLoopNum).HeatRecoveryBypass = true;
```

Comment:
Such a simple, straightforward change, yet it introduces max iteration warnings . . . ugh. Why? Investigating.

Reply:
Looking at detailed output with more output variables. The bypass flag is working as expected now, it's only 1 now when the OA mass flow is > the min OA (which can be reported using Air System Outdoor Air Mechanical Ventilation Requested Mass Flow Rate for this file anyway). So that much is ok.  But there are times when it is bouncing around from one HVAC timestep to the next, e.g. 01/02 08:00-10:00. During this time, the system flips betwen two states - HX active at min OA flow with a mixed air temp of 17C or so, or HX bypassed with OA flow > min OA and mixed air temps of 10C or less.  The times where it's bypassed, the OA flow must have been calculated with the HX active resulting in a warmer OA inlet node temp so the economizer opened up to reach the mixed air setpoint but then that caused the HX bypass to kick in and the next iteration comes around and the air is cold now.

*IF* this file had LockoutWithHeating instead of NoLockout, then the bouncy problem goes away. There's logic [here](https://github.com/NREL/EnergyPlus/blob/159313706_Issue%236796/src/EnergyPlus/MixedAir.cc#L4385-L4394) to avoid the no lockout nonsense we're seeing here. But this only executes for economizer LockoutWithHeating. Maybe this should execute also if BypassWhenOAFlowGreaterThanMinimum?

But even with LockoutWithHeating when the minimum OA is zero, then the MixedAirTempAtMinOAFlow = ReturnTemp which is 23C so the economizer thinks it should open up to reach 15.6C mixed air and then that triggers bypassing the HX which then results in a mixed air temp well below 15.6C. That's bad.

Is the [calculation of MixedAirTempAtMinOAFlow](https://github.com/NREL/EnergyPlus/blob/159313706_Issue%236796/src/EnergyPlus/MixedAir.cc#L3637-L3645) correct if there's an HX that could possibly be bypassed? Looking at the output for "Air System Outdoor Air Heat Recovery Bypass Minimum Outdoor Air Mixed Air Temperature" is seems like it's getting a cold value, but only when the min OA is > 0.

Thinking out loud here and documenting things for further review.

And this code is bad:
`this->MixedAirTempAtMinOAFlow = MixedAirTempAtMinOAFlow` - there shouldn't be a local MixedAirTempAtMinOAFlow declared earlier, delete that and this line.

Will revisit this tomorrow to propose a fix. Any suggestions from fresh eyes are welcome. @Myoldmopar @Nigusse

Reply:
@mjwitte Looking at the code MixedAirTempAtMinOAFlow is calculated using Outdoor Air Node (OANode) temperature [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/MixedAir.cc#L3639) where as the actual OA fraction (OutAirSignal) is calculated using mixer inlet node (InletNode, which is the heat recovery OA outlet node) temperature [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/MixedAir.cc#L4416). These temperature can be different depending on the status of the heat recovery HX. Could this be an cause for HX bypassed status flip?

Reply:
Yes, I think that's part of it. The thorny part is that the OA controller doesn't know what kind of equipment sits between OANode and InletNode, and it doesn't know if that equipment will be active or not. In the case of the HX, even that we don't know for sure that the HX Economizer Lockout = Yes. So it's not clear that `MixedAirTempAtMinOAFlow` could ever happen in some models. But for the original test file, this section doesn't execute because LockoutWithHeating is not set in the idf.

But for the test file *which does have LockoutWithHeating` the max iteration warnings are gone. It still doesn't do the right thing when minOA=0, but first we should get rid of the max iteration problem. 

The problem is likely that `InletTemp` will be different depending on whether the HXbypass flag was true on the prior iteration. There's some flag hocus pocus going on [here](https://github.com/NREL/EnergyPlus/blob/d7683729b22e5d468703c6d90789aa609a5f674e/src/EnergyPlus/MixedAir.cc#L3725-L3748) that's probably part of this.

Reply:
@Nigusse I tried moving the HX bypass check up before [this block](https://github.com/NREL/EnergyPlus/blob/159313706_Issue%236796/src/EnergyPlus/MixedAir.cc#L4541-L4613) which includes a re-simulation of the OA equipment. This got rid of the max iterations, but for some reason, the HX *never* goes into bypass with that. It may be necessary to call `SimOASysComponents` twice at this point (once with `HeatRecoveryBypass = true` and again with it `false`) and save the two InletTemp values and make a decision that way?  I'm fishing here for a robust solution.

Reply:
@mjwitte Comparison based on OA amount rather than fan fraction did not change the results for the test but sometimes it definitely avoids unnecessary SolveRoot call. The reason that the results did not change is that the HXBypass status is already captured by the original fix, I think. 

But I confirmed that MixedAirTempAtMinOAFlow calculated with and without HXBypass can be significantly different. For my test case I have seen (MixedAirTempAtMinOAFlow - MinedAirTemp) difference of as low as -2.5K deltaT when the HX is on and as high as 6.0K deltaT when the HX is bypassed.

Reply:
@Nigusse I've reverted my change, so things are back to the way you originally proposed. I will post a new issue regarding the max iterations. In many respects this bypass control doesn't make sense without also having LockoutWithHeating. That can be the workaround and can be explored in a reorg of this complicated controller which has many special parts all to fix various special problems.